apps/web/src/app/(signing)/sign/[token]/sign-dialog.tsx

```diff
@@ -13,30 +14,52 @@ export type SignDialogProps = {
   isSubmitting: boolean;
   document: Document;
   fields: Field[];
+  enableTips: () => void;
   onSignatureComplete: () => void | Promise<void>;
 };
 
 export const SignDialog = ({
   isSubmitting,
   document,
   fields,
+  enableTips,
   onSignatureComplete,
 }: SignDialogProps) => {
   const [showDialog, setShowDialog] = useState(false);
+  const [isFieldsSigned, setIsFieldsSigned] = useState(false);
 
   const isComplete = fields.every((field) => field.inserted);
 
+  const fieldSignCheck = () => {
+    const isAllFieldSigned = validateFieldsInserted(fields);
+    if (isAllFieldSigned) {
+      setIsFieldsSigned(true);
+    } else {
+    }
+  };
+
+  useEffect(() => {
+    if (fields.every((field) => field.inserted)) {
+      setIsFieldsSigned(true);
+    } else {
+      setIsFieldsSigned(false);
+    }
+  }, [fields]);
+
   return (
-    <Dialog open={showDialog} onOpenChange={setShowDialog}>
+    <Dialog open={isFieldsSigned && showDialog} onOpenChange={setShowDialog}>
       <DialogTrigger asChild>
         <Button
           className="w-full"
           type="button"
           size="lg"
-          disabled={!isComplete}
+          onClick={() => {
+            enableTips();
+            fieldSignCheck();
+          }}
           loading={isSubmitting}
         >
-          Complete
+          {isFieldsSigned ? <>Complete</> : <>Next Field</>}
```

Comment:
You should be able to use `isComplete` instead of `isFieldsSigned`

It's also preferred to just use text instead of fragments like this:

```
{isComplete ? 'Complete' : 'Next Field'}
```